scrapper.py

```diff
@@ -1,10 +1,140 @@
+
+    for each in text_data:
+        # parse price of flat
+        text_with_one_flat_price = ''.join(each.find('div',
+                                         class_='a-card__price').text.strip()[:-1].strip().split())
+        price = int(text_with_one_flat_price)
+        
+        # parse flat's square
+        text_with_one_flat_square = each.find('div',
+                                         class_='a-card__header-left').find('a',
+                                                                 class_='a-card__title').text
+        # search start and end indexes of pattern like: "'digits' м"
+        indexes_of_square_number = re.search(r"\d+(\.\d*)?\sм", text_with_one_flat_square) 
+        square = float(text_with_one_flat_square[indexes_of_square_number.start():
+                                                        indexes_of_square_number.end()-1])
+        # parse offer id and url to offer
+        text_with_one_offer_id = each.find('div',
+                                     class_='a-card__header-left').find('a',
+                                                                     class_='a-card__title').get('href')
+        url_id = f'{URL}{text_with_one_offer_id}'
+        # search  start and end indexes of digits in str
+        indexes_of_offer_id = re.search(r"\d+", text_with_one_offer_id)
+        id = int(text_with_one_offer_id[indexes_of_offer_id.start():
+                                                    indexes_of_offer_id.end()])
+
+        data[id] = (url_id, price, square)
+        
+    return len(data)
+
+def get_number_of_options(soup: BeautifulSoup) -> int:
+    """ Function parse scrapped BeautifulSoup object and return number of options."""
+    
+    text_with_numbers_of_options = soup.find('div',
+                                     class_='a-search-subtitle search-results-nb').text.strip().split()
+    numbers_of_options = int(text_with_numbers_of_options[1]) # number of options
+
+    return numbers_of_options
+
+def get_cost_per_sq_m(data: dict) -> dict:
+    """ Function get dictinary with parsed data and return dictionary format {id : int(cost per m^2)}."""
+        
+    cost_per_sq_m = {key: int(data[key][1]/data[key][2]) for key in data.keys()}
+    
+    return cost_per_sq_m
+
+
+
+def main():
+    """ Function scrap site by URL which entered in command line.
+        Result: dictinary with parsed data in format {offer id : tuple(url offer id, flat price, flat square)}
+        and dictinary with cost per m^2 in format {id : int(cost per m^2)}.
+    """
+    
+# Use this argument for test e.g.: 
+# https://krisha.kz/prodazha/kvartiry/almaty-aujezovskij-mkr-dostyk/?das[live.rooms]=1
+# https://krisha.kz/prodazha/kvartiry/almaty-aujezovskij-mkr-zhetysu-1/?das[live.rooms]=1 
+
+    command_line_arguments = argv #parse argument from CLI
+    page_number = int(1)
+    numbers_of_added_options = int(0)
+    numbers_of_options = int(1)
```

Comment:
@ProvVlad, why do you initiate this variable in a loop? You spend computational resources executing this initiation multiple times.

Reply:
@Shuk0, it's main function body, not a loop.